docs/data/date-pickers/custom-field/BrowserV7SingleInputRangeField.tsx

```diff
@@ -10,22 +10,14 @@ import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
```

Comment:
- Use `DateRangePickerFieldProps`
- Replace `BasePickersTextFieldProps` with `BaseSingleInputPickersTextFieldProps` (`BaseMultiInputPickersTextFieldProps` and `BaseSingleInputPickersTextFieldProps` are now incompatible so `BasePickersTextFieldProps` can't exist, and since we are migrating away from multi input I think we don't have to support this abstraction)
- Use `usePickersContext` to open the picker instead of passing a custom prop using `slotProps`